frontend/src/components/category-pill.tsx

```diff
@@ -4,16 +4,20 @@ import { twMerge } from 'tailwind-merge';
 interface CategoryPillProps extends React.HTMLAttributes<HTMLSpanElement> {
   category: string;
   selected?: boolean;
+  disabled?: boolean;
 }
 
-export default function CategoryPill({ category, selected = false }: CategoryPillProps) {
+
+export default function CategoryPill({ category, selected = false, disabled = false }: CategoryPillProps) {
   const [defaultColor, selectedColor]: [string, string] = getCategoryColors(category);
+  const disabledColor: string = "bg-slate-100 text-slate-400 cursor-not-allowed"
 
   return (
     <span
       className={twMerge(
         'cursor-pointer rounded-3xl px-3 py-1 text-xs font-medium text-light-primary/80 dark:text-dark-primary/80',
-        selected ? selectedColor : defaultColor
+        selected ? selectedColor : defaultColor,
+        disabled && disabledColor
       )}
```

Comment:
>disabled && disabledColor

I didn't get this logic, what is this? What are you trying to achieve? by this

> Then you can simply add disabled ? disabledColor : defaultColor to the twMerge() argument.
https://github.com/krishnaacharyaa/wanderlust/pull/69#issuecomment-1839463176

Why did you chose not to do this way? any  reason?

Reply:
Hey @chinmaykunkikar @krishnaacharyaa  here there is three state of the category
First is nomal state
second is selected state 
and third is disabled state
If i do like this
disabled ? disabledColor : defaultColor
how to handle selected state we need to like this
disabled ? disabledColor : selected ? selectedColor : defaultColor

Reply:
@SanjeevKrSingh
> disabled ? disabledColor : selected ? selectedColor : defaultColor

This seems reasonable to me.
But for better reading, we can split it to 
1. `disabled ? disabledColor : getSelectedColor(selected)`
OR
2. `getCategoryColor(disabled,selected)`
Inside `getCategoryColor` function, we can do `disabled ? disabledColor : getSelectedColor(selected)`

I find 2 modular and precise, and easy to understand, if you have any better approaches, feel free to let us know :) @chinmaykunkikar kindly share your thoughts if you get free time.